src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts

```diff
@@ -210,91 +225,38 @@ export class NotebookStickyScroll extends Disposable {
-			// check next cell, if markdown, that means this is the end of the section
-			// check if cell is within visible range
-			const nextCell = this.notebookEditor.cellAt(i + 1);
-			if (nextCell && i + 1 < visibleRange.end) {
-				if (nextCell.cellKind === CellKind.Markup) {
-					// this is the end of the section
-					// store the bottom scroll position of this cell
-					sectionBottom = this.notebookCellList.getCellViewScrollBottom(cell);
-					// compute sticky scroll height
-					const entry = NotebookStickyScroll.getVisibleOutlineEntry(i, this.notebookOutline.entries);
-					if (!entry) {
-						return;
-					}
-					// using 22 instead of stickyscrollheight, as we don't necessarily render each line. 22 starts rendering sticky when we have space for at least 1 of them
-					const newStickyHeight = NotebookStickyScroll.computeStickyHeight(entry!);
-					if (editorScrollTop + newStickyHeight < sectionBottom) {
-						trackedEntry = entry;
-						break;
-					} else {
-						// if (editorScrollTop + stickyScrollHeight > sectionBottom), then continue to next section
-						continue;
-					}
-				}
-			} else {
-				// there is no next cell, so use the bottom of the editor as the sectionBottom, using scrolltop + height
-				sectionBottom = this.notebookEditor.scrollTop + this.notebookEditor.getLayoutInfo().scrollHeight;
-				trackedEntry = NotebookStickyScroll.getVisibleOutlineEntry(i, this.notebookOutline.entries);
-				break;
-			}
-		} // cell loop close
-
-		// -------------------------------------------------------------------------------------
-		// we now know the cell which the sticky is determined by, and the sectionBottom value to determine how many sticky lines to render
-		// compute the space available for sticky lines, and render sticky lines
-
-		const linesToRender = Math.floor((sectionBottom - editorScrollTop) / 22);
-		let newMap: Map<OutlineEntry, { line: NotebookStickyLine; rendered: boolean }> = new Map();
-		newMap = NotebookStickyScroll.renderStickyLines(trackedEntry?.parent, this.domNode, linesToRender, newMap, this.notebookEditor);
-		this.setCurrentStickyLines(newMap);
-		this.updateDisplay();
-	}
+		this.disposeCurrentStickyLines();
+		this.renderStickyLines(newMap, this.domNode);
 
-	private updateContent(newMap: Map<OutlineEntry, { line: NotebookStickyLine; rendered: boolean }>) {
+		const oldStickyHeight = this.getCurrentStickyHeight();
 		this.setCurrentStickyLines(newMap);
+		this.renderedStickyLines = Array.from(newMap.values())
+			.filter(value => value.rendered)
+			.map(value => value.line);
+
+		// (+) = sticky height increased
+		// (-) = sticky height decreased
+		const sizeDelta = this.getCurrentStickyHeight() - oldStickyHeight;
+		if (sizeDelta !== 0) {
+			this._onDidChangeNotebookStickyScroll.fire(sizeDelta);
+		}
 		this.updateDisplay();
 	}
 
 	private updateDisplay() {
-		const hasSticky = this.currentStickyLines.size > 0;
+		const hasSticky = this.getCurrentStickyHeight() > 0;
```

Comment:
we can check the size of `currentStickyLines`

Reply:
Don't think so, as that is all of the potential lines to be rendered for a section. As they collapse against the next section, some will not be rendered. Need to specifically check with the fxn since it checks the rendered value